src/lwc/utilTemplateBuilder/utilTemplateBuilder.js

```diff
@@ -315,13 +379,20 @@ const showToast = (title, message, variant, mode, messageData) => {
 }
 
 /*******************************************************************************
-* @description Creates and dispatches an error toast.
+* @description Creates and dispatches an error toast or dispatches an application
+ * event to be handle the error gracefully
 *
 * @param {object} error: Event holding error details
 */
-const handleError = (error) => {
-    let message = commonUnknownError;
-
+const handleError = async (error) => {
```

Comment:
I would say we should take a step back here and reconsider how this method is written and being used.  The code in this block is growing significantly and .  Maybe we can have a conversation about how to refactor and move fwd with this function.